sparta/src/Scheduler.cpp

```diff
@@ -108,7 +108,12 @@ Scheduler::Scheduler(const std::string& name, GlobalTreeNode* search_scope) :
     // start/stop times for differencing.
     sset_.setClock(scheduler_internal_clk_.get());
 
+    // TODO NOTE: Should we turn cycle-checking on by default (true)?
+    // Cycle-checking tells the DAG to flag a cycle at the earliest opportunity
+    // This means checking for a cycle as each edge is added to the DAG (DAG::link)
+    // It's expensive, but can save time in debugging precedence problems
```

Comment:
YES!  However, if you're concerned about the time spend, we can add it as a boolean to the `SimulationConfiguration` class and pass that value through to the Scheduler...

Reply:
that would be great!

Reply:
Do you want me to do that in this branch or after the merge of the PR?  I think the latter...

Reply:
Yeah, if it's ok to wait for the PR merge, then you can just do in master or another tiny branch

Reply:
(Guess we need another approver to sign off first, though)